### ISSUE TITLE
how to apply FSAM with gradient accumulation training 

### ISSUE THREAD

**Issue by stylebased** [nblt/F-SAM#3]
Hi,
Thank you for sharing your impressive work. Could you please provide the code for using your FSAM with gradient accumulationin training?
Best

**Comment by nblt** [comment]
Hi,

Thank you for your question. For gradient accumulation training, we need to accumulate the flat gradient during the second forward-backward step. Please note that we have to store a copy of the accumulated gradients, as we will clear the gradients during the SAM's first gradient step.

I've uploaded an example usage in the [grad_accumulate](https://github.com/nblt/F-SAM/tree/grad_accumulate) branch for your reference.

Best

**Comment by stylebased** [comment]
thanks
